value-processor/src/org/immutables/value/processor/meta/ValueAttribute.java

```diff
@@ -1136,14 +1136,19 @@ void initAndValidate(@Nullable InstantiationCreator instantiationCreator) {
     validateThrowsClause();
     validateTypeAndAnnotations();
 
+    initAttributeValueType();
```

Comment:
putting this into `supportBuiltinContainerTypes()` was probably something short-sighted, but it should work correctly here too

Reply:
@elucash thank your for looking into it.
Is it OK if I change `ValueAttribute` ? 
The thing I was missing is :
```java
// inside initAttributeValueType
|| containingType.isGenerateCriteria()
```

Reply:
yep, `ValueAttribute` changes should be ok, the other part is incomplete